vowpalwabbit/csoaa.cc

```diff
@@ -35,7 +35,7 @@ inline void inner_loop(single_learner& base, example& ec, uint32_t i, float cost
 {
   if (is_learn)
   {
-    ec.weight = (cost == FLT_MAX) ? 0.f : 1.f;
+    (cost == FLT_MAX) ? ec.weight = 0.0f : ec.weight = 1.0f;
```

Comment:
We should just remove this line instead, right?  Overriding the weight directly invalidates setting the weight at higher levels.